contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -0,0 +1,280 @@
+        for (uint i = 0; i < messages.length; i++) {
+            priceData.put(messages[i], signatures[i]);
+        }
+
+        uint ethPrice = fetchEthPrice();
+
+        // Try to update the view storage
+        for (uint i = 0; i < symbols.length; i++) {
+            TokenConfig memory config = getTokenConfigBySymbol(symbols[i]);
+            string memory symbol = symbols[i];
+            bytes32 symbolHash = keccak256(abi.encodePacked(symbol));
+            if (source(messages[i], signatures[i]) != reporter) continue;
+
+            uint reporterPrice = priceData.getPrice(reporter, symbol);
+            uint anchorPrice;
+            if (symbolHash == ethHash) {
+                anchorPrice = ethPrice;
+            } else {
+                anchorPrice = fetchAnchorPrice(config, ethPrice);
+            }
+
+            uint anchorRatio = mul(anchorPrice, 100e16) / reporterPrice;
+            bool withinAnchor = anchorRatio <= upperBoundAnchorRatio && anchorRatio >= lowerBoundAnchorRatio;
+
+            if (reporterInvalidated == true) {
+                if (prices[symbolHash] == anchorPrice) {
+                    prices[symbolHash] = anchorPrice;
+                }
+            } else if (withinAnchor) {
+                if (prices[symbolHash] != reporterPrice) {
+                    prices[symbolHash] = reporterPrice;
+                    emit PriceUpdated(symbol, reporterPrice);
+                }
+            } else {
+                emit PriceGuarded(symbol, reporterPrice, anchorPrice);
+            }
+        }
+    }
+
+    /**
+     * @dev Fetches the current token/eth price accumulator from uniswap.
+     */
+    function currentCumulativePrice(TokenConfig memory config) internal view returns (uint) {
+        (uint cumulativePrice0, uint cumulativePrice1,) = UniswapV2OracleLibrary.currentCumulativePrices(config.uniswapMarket);
+        if (config.isUniswapReversed) {
+            return cumulativePrice1;
+        } else {
+            return cumulativePrice0;
+        }
+    }
+
+    /**
+     * @dev Fetches the current eth/usd price from unsiwap, with 6 decimals of precision.
+     *  Conversion factor is 1e18 for eth/usdc market, since we decode uniswap price statically with 18 decimals.
+     */
+    function fetchEthPrice() internal returns (uint) {
+        return fetchAnchorPrice(getTokenConfigBySymbolHash(ethHash), 1e18);
+    }
+
+    /**
+     * @dev Fetches the current token/usd price from uniswap, with 6 decimals of precision.
+     */
+    function fetchAnchorPrice(TokenConfig memory config, uint conversionFactor) internal virtual returns (uint) {
+        (uint nowCumulativePrice, uint oldCumulativePrice, uint oldTimestamp) = pokeWindowValues(config);
+        uint timeElapsed = block.timestamp - oldTimestamp;
+
+        // Calculate uniswap time-weighted average price
+        FixedPoint.uq112x112 memory priceAverage = FixedPoint.uq112x112(uint224((nowCumulativePrice - oldCumulativePrice) / timeElapsed));
```

Comment:
Can timeElapsed be zero? What does it mean here?

Reply:
Didn't you prove its >= period?